src/arch/arm/linux/linux.hh

```diff
@@ -568,6 +569,45 @@ class ArmLinux64 : public ArmLinux, public OpenFlagTable<ArmLinux64>
         uint64_t cgroup;
     };
 
+    // For sigreturn
+    struct tgt_sigframe
+    {
+        struct tgt_siginfo
+        {
+            int si_signo;
+            int si_errno;
+            int si_code;
+            // union sigval... Not used by us. Pad to correct size instead.
+            unsigned char _pad[128 - 3 * sizeof(int)];
+        } info;
+
+        struct tgt_ucontext
+        {
+            unsigned long int uc_flags;
+            void *uc_link; // not used by us
+            struct
+            {
+                void *ss_sp;
+                size_t ss_size;
+                int ss_flags;
+            } uc_stack;
+            unsigned char uc_sigset_ex[136];
+            struct
+            {
+                unsigned long long int fault_address;
+                unsigned long long int regs[31];
+                unsigned long long int sp;
+                unsigned long long int pc;
+                unsigned long long int pstate;
```

Comment:
Any reason not to use `uint64_t` here? I believe that the C standard states `long long int` guarantees a value of _at least_ 64 bits while `uint64_t` guarantees _exactly_ 64 bits. Ergo, the latter is more portable.

I also think `uint64_t` is easier to read. I always need a minute to remind myself what `long long` means.

Reply:
That's a good catch @BobbyRBruce .

I will add this is not just for the sake of readability. If we run gem5 on a 32bit system, we would get the wrong tgt_struct size when trying to emulate an Arm64 SE system

Reply:
Thanks, great suggestion. Done.